app/webpacker/controllers/order_cycle_controller.js

```diff
@@ -0,0 +1,52 @@
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  static targets = ['input', 'modalConfirm'];
+  static values = { initVals: { type: Object, default: {} }, hasSchedule: { type: Boolean, default: false } };
+
+  connect() {
+    if(!this.hasScheduleValue) return;
+    // Attach update callback method
+    window.adminOrderCycleUpdateCallback = this.updateCallback.bind(this);
+  }
+
+  toggleSaveBtns() {
+    if(!this.hasScheduleValue) return;
+
+    // Check that datetime input value has a change
+    const dirty = this.inputTargets.some(ele =>
+      new Date(this.initValsValue[`${ele.name}`]).getTime() !== new Date(ele.value).getTime());
+
+    // Toggle save bar action button
+    if (dirty) {
+      this.element.querySelector('#form-actions').style.display = 'none';
+      this.element.querySelector('#modal-actions').style.display = 'unset';
+    } else {
+      this.element.querySelector('#form-actions').style.display = 'unset';
+      this.element.querySelector('#modal-actions').style.display = 'none';
+    }
+  }
+
+  updateModalConfirmButton(e) {
+    if(!this.hasScheduleValue) return;
+    // Display modal confirm button coresponding to save bar button clicked
+    this.modalConfirmTargets.forEach(ele => {
+      if (e.target.getAttribute('data-target') === ele.getAttribute('data-request')) {
+        ele.style.display = 'unset';
+      } else {
+        ele.style.display = 'none';
+      }
+    });
```

Comment:
It's a bit hard to follow what's happening, but if understand correctly you are using the `data-target` to display the correct button the modal, right ?
I think you could just pass a parameter to  `updateModalConfirmButton()` to achieve the same thing, and I believe it would make the intent of the function easier to understand. 
Something like : 

```suggestion 
    this.modalConfirmTargets.forEach(ele => {
      if (e.params.request === ele.getAttribute('data-request')) {
        ele.style.display = 'unset';
      } else {
        ele.style.display = 'none';
      }
    });
```
Then you can call it like so :
```
 %input.red{ type: "button", value: t('.save_and_next'), "ng-disabled": "!order_cycle_form.$dirty || order_cycle_form.$invalid", data: { 'action': 'click->modal-link#open click->order-cycle#updateModalConfirmButton', 'order-cycle-request-param': 'saveAndNext'} }
```